src/System.Memory/src/System/Buffers/ReadOnlySequence.cs

```diff
@@ -37,19 +37,16 @@ namespace System.Buffers
         /// <summary>
         /// Determines if the <see cref="ReadOnlySequence{T}"/> contains a single <see cref="ReadOnlyMemory{T}"/> segment.
         /// </summary>
-        public bool IsSingleSegment => _sequenceStart.GetObject() == _sequenceEnd.GetObject();
+        public bool IsSingleSegment
+        {
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            get => _sequenceStart.GetObject() == _sequenceEnd.GetObject();
+        }
 
         /// <summary>
         /// Gets <see cref="ReadOnlyMemory{T}"/> from the first segment.
         /// </summary>
-        public ReadOnlyMemory<T> First
-        {
-            get
-            {
-                TryGetBuffer(_sequenceStart, _sequenceEnd, out ReadOnlyMemory<T> first, out _);
-                return first;
-            }
-        }
+        public ReadOnlyMemory<T> First => GetFirstBuffer(_sequenceStart, _sequenceEnd);
```

Comment:
Why does `CheckEndReachable` take arguments?

Reply:
Move it up into caller because only 2 of the various Slices use it; is now a `static` method and needs to know what start and end to use (as they aren't instance values)